osrsbox/items_api/all_items.py

```diff
@@ -0,0 +1,118 @@
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+###############################################################################
+"""
+
+import os
+import json
+import glob
+from typing import Dict
+from typing import List
+from typing import Generator
+
+from osrsbox.items_api import item_definition
+
+
+class AllItems:
+    """This class handles loading of the osrsbox-db items database.
+
+    :param input_data_file_or_directory: The osrsbox-db items folder of JSON files, or single JSON file.
+    """
+    def __init__(self, input_data_file_or_directory: str):
+        self.all_items: List[object] = list()
+        self.all_items_dict: Dict[int, object] = dict()
+        self.load_all_items(input_data_file_or_directory)
+
+    def __iter__(self) -> Generator[str, None, None]:
+        """Iterate (loop) over each ItemDefinition object."""
+        for item_id in self.all_items:
+            yield item_id
+
+    def __getitem__(self, id_number: int) -> object:
+        """Return the item definition object for a loaded item.
+
+        :param id_number: The item ID number.
+        :return: The item definition object linked to a specific ID number.
+        """
+        return self.all_items_dict[id_number]
+
+    def __len__(self) -> int:
+        """Return the count of the total number of items.
+
+        :return: The total number of items.
+        """
+        return len(self.all_items)
+
+    def load_all_items(self, input_data_file_or_directory: str):
+        """Load the items database via a JSON file, or directory of JSON files.
+
+        :param input_data_file_or_directory: The path to the data input.
+        """
+        if os.path.isdir(input_data_file_or_directory):
+            if input_data_file_or_directory.endswith("/"):
+                path = input_data_file_or_directory + "*"
+            else:
+                path = os.path.join(input_data_file_or_directory, "*")
+
+            self._load_items_from_directory(path_to_directory=path)
+
+        elif os.path.isfile(input_data_file_or_directory):
+            self._load_items_from_file(input_data_file_or_directory)
+
+    def _load_items_from_directory(self, path_to_directory: str):
+        """Load item database from a directory of JSON files (`items-json`).
+
+        :param path_to_directory: The path to the `items-json` directory.
+        """
+        count = 1
+        # Loop through every item file
+        for json_file in glob.glob(path_to_directory):
+
+            if os.path.isdir(json_file):

```

Comment:
One interesting thing that you can do is to turn this condition as part of the generator so that you don't run into it and clear up your logic.

```python
json_files = (json_file for json_file in glob.glob(path_to_directory) if not os.path.isdir(json_file))
for json_file in json_files:
    # You don't have to do that check anymore!
```
Some people like this style, just wanted you to know an alternative.